internal/resources/slo/resource_slo.go

```diff
@@ -338,6 +365,40 @@ func packQuery(query map[string]interface{}) (gapi.Query, error) {
 		return sloQuery, nil
 	}
 
+	if query["type"] == "ratio" {
+		ratioquery := query["ratio"].([]interface{})[0].(map[string]interface{})
+		successMetric := ratioquery["success_metric"].(string)
+		totalMetric := ratioquery["total_metric"].(string)
+		groupByLabels := ratioquery["group_by_labels"].([]interface{})
+
+		var labels []string
+
+		for ind := range groupByLabels {
+			if groupByLabels[ind] == nil {
```

Comment:
you might as well drop any empty-string labels (i.e. length < 1). The backend will do the same, but only cause the FE had a similar bug. We might in the future,  start rejecting SLOs because empty-string isn't a valid prometheus label name

Reply:
not a big deal, but would be better not to send emptystrings in the list